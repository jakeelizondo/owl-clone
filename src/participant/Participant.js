import React from 'react';
import './Participant.css';

function Participant(props) {
  return (
    <div className="participant">
      {!props.invertImage ? (
        <div>
          <img src={props.avatar} />
          <h2>{props.name}</h2>
          <p>{props.onStage ? 'on stage' : null}</p>
        </div>
      ) : (
        <div>
          <h2>{props.name}</h2>
          <img src={props.avatar} />
        </div>
      )}
    </div>
  );
}

export default Participant;
