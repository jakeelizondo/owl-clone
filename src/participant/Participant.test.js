import React from 'react';
import ReactDOM from 'react-dom';
import Participant from './Participant';
import renderer from 'react-test-renderer';
describe('Participant Component', () => {
  // smoke test for app component

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Participant />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  // snapshot test
  it('renders the UI as expected', () => {
    const tree = renderer.create(<Participant />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
