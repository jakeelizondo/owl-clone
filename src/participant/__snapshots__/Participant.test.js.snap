// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Participant Component renders the UI as expected 1`] = `
<div
  className="participant"
>
  <div>
    <img />
    <h2 />
    <p />
  </div>
</div>
`;
